packages/pie-toolbox/src/code/editable-html/editor.jsx

```diff
@@ -420,18 +421,74 @@ export class Editor extends React.Component {
   };
 
   onEditingDone = () => {
-    if (this.state.isHtmlMode) {
+    const { isHtmlMode, dialog, value, pendingImages } = this.state;
+
+    // Handling HTML mode and dialog state
+    if (isHtmlMode && !dialog?.open) {
+      const currentValue = htmlToValue(value.document.text);
+      const previewText = this.renderPreviewText(currentValue.document.text);
+
+      this.openDialogWithPreview(currentValue, previewText);
       return;
     }
 
-    const { pendingImages } = this.state;
+    // Early return if HTML mode is enabled
+    if (isHtmlMode) return;
 
+    // Handling pending images
     if (pendingImages.length) {
-      this.setState({ scheduled: true });
+      this.scheduleImageProcessing();
       return;
     }
 
-    log('[onEditingDone]');
+    // Finalizing editing
+    this.finalizeEditing();
```

Comment:
Same here. I agree it looks nice to have specific functions to do stuff, but it would be easier to read if we keep it as it was.